### ISSUE TITLE
Accessibility in Russia

### ISSUE THREAD

**Issue by DoctorRyner** [elm/package.elm-lang.org#300]
We're terribly sorry but our government banned this IP in attempts to block the telegram messenger, that's funny isn't it? I mean telegram is still working though ... but I can't say the same about this and now we have to use VPN, it's pretty tough and inconvenient. So... we would love if you would do something about that like transferring it to an another IP and if you would then we will be happy to support you with our donations

**Comment by VladimirLogachev** [comment]
I can donate $100 for that

**Comment by bobwhitelock** [comment]
Looks like this is a duplicate of https://github.com/elm/package.elm-lang.org/issues/278. You could be waiting for a while for this to be fixed though, that issue's been open 6 months and no one with the power to change this appears to have responded or taken any action 🤷‍♂️

**Comment by VladimirLogachev** [comment]
This issue means that we have money and problem.
And we need `package.elm.org` administrator to accept money and solve a problem.

**Comment by VladimirLogachev** [comment]
@evancz what do you think about our problem with Russian government?
can we solve it by moving server to another IP address?

**Comment by dnsl48** [comment]
@VladimirLogachev did you consider to write official requests to roskomnadzor? https://rkn.gov.ru/treatments/
AFAIK, they have to officially respond to such requests on paper (as long as you send them a request on paper).
If they got tons of such requests on paper, they might get bored to respond and unblock this particular IP address?

**Comment by VladimirLogachev** [comment]
Of course, we try to communicate with them.
They say thad DigitalOcean should obey Roskomnadzor (RKN) in every case, not just this IP or Telegram, and then RKN will unblock it so we could access package server.

Even if we ask, there is no reason for DigitalOcean to agree to obey RKN.
But we wouldn't, because neither RKN nor any government structure is not our master, even though they have some force and impact.

It's not only a Russian problem - blocking things on the internet is our new reality. Today - Russia, tomorrow - China, or some other eastern country.

Take a look at [geography of Elm language on Google Trends](https://trends.google.com/trends/explore?q=%2Fm%2F0ncc1sv). Not all of these governments are sane and allow information to flow freely. In a near perspective we need all critical things to be available via a specified mirror or proxy.

**Comment by dnsl48** [comment]
Yeah, sorry, I didn't mean to ask them, but rather flooding them with official requests as a form of civil protest. Not saying it must help though, but maybe worth considering a try?

Even though I'm not in any way involved in the elm infrastructure project, I still don't believe changing IP address is the best approach here. You just said yourself:

> It's not only a Russian problem - blocking things on the internet is our new reality. Today - Russia, tomorrow - China, or some other eastern country.

That would just generate tons of toil for infrastructure admins and is merely a temporary solution anyway.

I would guess a better way would rather be to contribute some sort of server mirrors support for elm package manager. And then potentially maintain a Russian CDN or at least maybe a mirror?
Just IMHO here.

**Comment by DoctorRyner** [comment]
> is merely a temporary solution anyway
Well yes, but actually no :)
I mean they do not in a pursue for blocking Elm or anything, they just don't want to admit their mistake and fix it, I doubt that it will be blocked again that way any time soon

**Comment by DoctorRyner** [comment]
> tons of toil for infrastructure admins
I doubt that it would be nearly that hard and even if it would we will suggested to compensate it

**Comment by rlefevre** [comment]
Until an official solution is found, you can use the elm-package.org proxies described in [this gist](https://gist.github.com/rlefevre/7c21513c3cc14ea23524223ab35e4ca6#file-elm-dmy-fr-md). It should be easier to use than a VPN, particularly for CI.

**Comment by AleXoundOS** [comment]
There needs to be an official guide to configuring the proxy and connection to hypothetical mirrors.

**Comment by rlefevre** [comment]
I have also added a reverse proxy to browse documentation for packages < 0.19:
https://old.elm.dmy.fr/

To use 0.18 `elm-make` and `elm-package`, use:
```
$ http_proxy=elm.dmy.fr:9999 elm-make [...]
$ http_proxy=elm.dmy.fr:9999 elm-package ...
```

**Comment by rlefevre** [comment]
From https://github.com/elm/package.elm-lang.org/issues/278#issuecomment-562859978
> The site has a new IP address. Is it working for people in Russia now?

**Comment by DoctorRyner** [comment]
Yes, it seems the problem is gone
